docs/contributing/core/index.md

```diff
@@ -0,0 +1,268 @@
+```
+
+The current default and development branch of `buildout.coredev` is `6.1`.
+Older versions are named according to their `major.minor` version.
+
+Always begin by checking out the git branch on which you want to work, usually called `origin`.
+If you have not yet checked out a branch, then you need to track it with the `-t` option and specify the remote branch to track.
+The following command will switch and track changes from the remote branch `origin/6.1`.
+
+```shell
+git checkout -t origin/6.1
+```
+
+If you check out a different branch, and want to return to the previous branch, you need to specify only the local branch.
+
+```shell
+git checkout 6.1
+```
+
+Next pull down and merge any recent changes from the remote tracked repository with a single command.
+
+```shell
+git pull
+```
+
+```{important}
+Make sure to rerun buildout for the current branch to get the correct versions of packages, otherwise you will get some weird behavior.
+```
+
+Next create a new branch on which you want to work from the current branch, tracking the upstream Plone repository, and check it out.
+It's a good idea to use a branch name that includes the issue number and is descriptive of what it resolves.
+
+```shell
+git switch -c 123-thing-i-fixed -t origin/6.1
+```
+
+Now you can edit your code without affecting the original branch.
+
+
+## Edit packages
+
+First identify the names of the Plone packages you want to work on.
+If you do not know, you can ask in the [Plone Community Forum](https://community.plone.org/).
+Only a few packages are in {file}`src/` by default.
+
+Edit the file {file}`checkouts.cfg` at the root of the repository by adding the package names under the `auto-checkout` list.
+
+```{code-block} cfg
+:emphasize-lines: 10-
+[buildout]
+always-checkout = force
+# always-checkout = false
+auto-checkout =
+# These packages are always checked out:
+    docs
+    # <snip>
+# These packages are manually added, or automatically added by mr.roboto:
+    # <snip>
+    plone.app.event
+    icalendar
+```
+
+Then rerun buildout to install the source packages to edit.
+
+```shell
+./bin/buildout
+```
+
+```{seealso}
+For tips on working with `mr.developer`, see {doc}`mrdeveloper`.
+```
+
+
+## Test locally
+
+If you change the expected behavior of a feature in a package, you should write a test to cover the change.
+
+To run a test for the specific package that you modified, use the `-m` option followed by the package name, as shown in the following example.
+
+```shell
+./bin/test -m plone.app.caching
+```
+
+If any test fails, do not commit and push the changes.
+Instead write a test that passes.
+
+After the package level tests pass with your change, you can run the full unit test suite to ensure other packages aren't affected by the change.
+It takes 5-10 minutes to run the full unit test suite.
```

Comment:
You can do this, but these days we have CI, so usually I create a PR at this point and ask Jenkins to run the tests.

Reply:
Done in fcdd9923dcb60a1007cea70e2ebf9c3bdf73544d